contracts/policy-agents/PolicyOfferExecutor.sol

```diff
@@ -0,0 +1,129 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../interfaces/IClaimsManager.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract PolicyOfferExecutor {
```

Comment:
I know this is draft PR but we are going to create an interface for this contract, correct?

Reply:
Also, would it be worth adding an event to the executeOffer() functions or should we just rely on the underlying events emitted by ClaimsManager?